prospector/api/main.py

```diff
@@ -89,7 +89,7 @@ async def create_data(repository_url, commit_id, label, vulnerability_id):
 @app.get("/commits/{repository_url}")
 # async def get_commits(repository_url, commit_id=None, token=Depends(oauth2_scheme)):
 async def get_commits(repository_url, commit_id=None):
-    commit = (commit_id, repository_url)
+    commit = {"commit_id": commit_id, "repository_url": repository_url}
```

Comment:
I think I preferred the more concise version which was as readable and, with type hints, still easy to develop using IDE support

Reply:
I modified it to be a dict on processing side to make confusion of parameters impossible.

Reply:
BTW: i think this should fail if commit_id is None (is that the reason why this commit fails pytest?)

Reply:
> Thanks @copernico, I still need to figure out how a test broke
Do let me know if you need help with that

Reply:
Actually I could not make debugging work. The `uvicorn api.main:app --reload ` command works, however if I translate this to a lauch.json as: ```
{
    "version": "0.2.0",
    "configurations": [
        
        {
            "name": "Python: FastAPI",
            "type": "python",
            "request": "launch",
            "module": "uvicorn",
            "args": [
                "api.main:app", "--reload"
            ],
            "jinja": true,
            "cwd": "${workspaceFolder}",
            "envFile": "${workspaceFolder}/.env"
        }
    ]
} 
```
The I get "No module named uvicorn" error. Have you encountered such issue?

Reply:
No, I start it in a separate terminal and keep it running; it reloads itself automatically if I change any file, so no need to worry about starting it explicitly all the time

Reply:
as for the failing tests, this one:
https://github.com/SAP/project-kb/pull/88/checks?check_run_id=2342647297#step:7:71
depends on the use of commit as a dictionary I guess.
The other one is a mistake that I introduced (and i will fix in a sec)

Reply:
> No, I start it in a separate terminal and keep it running; it reloads itself automatically if I change any file, so no need to worry about starting it explicitly all the time

Yes, for that purpose it works great. It would be also great if i could insert break points to debug it on the fly.